backend/src/http/handlers.rs

```diff
@@ -382,7 +382,23 @@ pub(super) trait CommonHeadersExt {
 }
 
 impl CommonHeadersExt for hyper::http::response::Builder {
-    fn with_content_security_policies(self, _config: &Config, _nonce: &str) -> Self {
+    fn with_content_security_policies(self, config: &Config, _nonce: &str) -> Self {
+        let redirect_actions = if config.auth.pre_auth_external_links {
+            [config.opencast.studio_url(), config.opencast.editor_url()]
+                .into_iter()
+                .map(|uri| {
+                    let mut parts = uri.0.into_parts();
+                    parts.path_and_query = Some(PathAndQuery::from_static("/redirect/get"));
+                    Uri::from_parts(parts).unwrap().to_string()
+                })
+                .collect::<HashSet<_>>()
+                .into_iter()
+                .collect::<Vec<_>>()
+                .join(" ")
```

Comment:
The fact that there is still no `Iterator::join` ... grumble. There is the unstable `intersperse` which is cool in its own right, but still will be annoying because all items will need to have the same type. Nothing to change here, just ranting.

Reply:
Ye it's unfortunate. 😞 

This code went through a few iterations, too. Seems very overkill to me just to make sure we don't duplicate a `form-action`. But oh well, it is what it is. 🤷‍♀️